File: src/main/java/no/nav/samordning/innlastning/database/DatasourceConfig.java
package no.nav.samordning.innlastning.database;

import com.zaxxer.hikari.HikariConfig;

public class DatasourceConfig {

    private static final int MIN_IDLE = 0;
    private static final int MAX_LIFETIME_MS = 30001;
    private static final int MAX_POOL_SIZE = 3;
    private static final int CONNECTION_TIMEOUT_MS = 250;
    private static final int IDLE_TIMEOUT_MS = 10001;

    public static HikariConfig getDatasourceConfig(String dbUrl) {
        HikariConfig datasourceConfig = new HikariConfig();
        datasourceConfig.setJdbcUrl(dbUrl);
        datasourceConfig.setMinimumIdle(MIN_IDLE);
        datasourceConfig.setMaxLifetime(MAX_LIFETIME_MS);
        datasourceConfig.setMaximumPoolSize(MAX_POOL_SIZE);
        datasourceConfig.setConnectionTimeout(CONNECTION_TIMEOUT_MS);
        datasourceConfig.setIdleTimeout(IDLE_TIMEOUT_MS);
        return datasourceConfig;
    }
}

File: src/main/kotlin/no/nav/samordning/innlasting/database/DatasourceConfig.kt
package no.nav.samordning.innlasting.database

import com.zaxxer.hikari.HikariConfig

object DatasourceConfig {

    private const val MIN_IDLE = 0
    private const val MAX_LIFETIME_MS = 30001L
    private const val MAX_POOL_SIZE = 3
    private const val CONNECTION_TIMEOUT_MS = 250L
    private const val IDLE_TIMEOUT_MS = 10001L

    fun getDatasourceConfig(dbUrl: String) = HikariConfig().apply {
        jdbcUrl = dbUrl
        minimumIdle = MIN_IDLE
        maxLifetime = MAX_LIFETIME_MS
        maximumPoolSize = MAX_POOL_SIZE
        connectionTimeout = CONNECTION_TIMEOUT_MS
        idleTimeout = IDLE_TIMEOUT_MS
    }
}
